gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -195,6 +207,25 @@ QueryNode buildExecutionTree(ExecutionContext executionContext, QueryNode filter
     return rootNode;
   }
 
+  @VisibleForTesting
+  QueryNode buildFilterTree(ExecutionContext context, Filter filter) {
+    // Convert the time range into a filter and set it on the request so that all downstream
+    // components needn't treat it specially
+    Filter timeRangeFilter =
+        TimeRangeFilterUtil.addTimeRangeFilter(
+            context.getTimestampAttributeId(),
+            filter,
+            context.getEntitiesRequest().getStartTimeMillis(),
+            context.getEntitiesRequest().getEndTimeMillis());
+
+    return buildFilterTree(
```

Comment:
Doesn't `TimeRangeFilterUtil.addTimeRangeFilter` already AND the input filter?
Do we need to explicitly create another filter?

Reply:
Oh right